### PR TITLE
A few small fixes to errors, warnings, and notes from `devtools::check()`

### PR DESCRIPTION
**Pull request by ignacio82**
I'm submitting this pull request in case you find it useful. I am not able to install this package in its current state. Here is a quick way of reproducing the error that I had:

1. Create a Dockerfile with just the following:

```
FROM rocker/r-ubuntu:18.04
# Install dependencies
RUN install.r remotes \
    && installGithub.r mathematica-mpr/bcf-1
```

All this is doing is grabbing a clean version of r and installing the remotes package and your bcf package.

2. Build that container by running the following:

```
$ docker build --tag bcf .
```

Alas, you get the following error:

```
....
Error: package or namespace load failed for ‘bcf’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/usr/local/lib/R/site-library/00LOCK-bcf/00new/bcf/libs/bcf.so':
  /usr/local/lib/R/site-library/00LOCK-bcf/00new/bcf/libs/bcf.so: undefined symbol: dgesvx_
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/usr/local/lib/R/site-library/bcf’
Error: Failed to install 'bcf' from GitHub:
  (converted from warning) installation of package ‘/tmp/file3bf4b2624/bcf_2.0.0.tar.gz’ had non-zero exit status
```

I fixed this and a few other very minor things in this pull request. You can verify it by changing the Dockerfile to `ignacio82/bcf` instead of  `mathematica-mpr/bcf-1`

That said, there are still many things that you will have to fix before you can actually publish this on cran. You should try running `devtools::check()` to get a list of things to fix.

### CODE REVIEW

**Comment by ignacio82** [comment]
you should probably run `rhub::check()` to make sure the package can be built in different platforms (windows, mac, and linux).

File: DESCRIPTION [hunk 1]
```diff
@@ -20,12 +20,29 @@ RoxygenNote: 7.0.2
...
+    latex2exp,
+    ggplot2,
+    rpart,
+    rpart.plot
 SystemRequirements: GNU make
 Language: en-US
 VignetteBuilder: knitr
+Remotes: 
+    mathematica-mpr/coda
+Encoding: UTF-8
```
Comment (TychonautVII) [inline DESCRIPTION]: @ignacio82 Was the "encoding" line what ended up fixing your problem? I see an "undefined" symbol error.

I'm wondering what fixed you're bug vs what was general fixes for devtools:check()

File: src/Makevars [hunk 2]
```diff
@@ -1,5 +1,7 @@
-
-## Use the R_HOME indirection to support installations of multiple R version
 CXX_STD = CXX11
-PKG_LIBS += $(shell ${R_HOME}/bin/Rscript -e "RcppParallel::RcppParallelLibs()")
+PKG_CXXFLAGS += -DRCPP_PARALLEL_USE_TBB=1
+
+PKG_LIBS = $(shell "${R_HOME}/bin/Rscript" -e  -e "RcppParallel:::LdFlags()") $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)
 
+PKG_LIBS += $(shell "${R_HOME}/bin/Rscript" -e  \
+              -e "RcppParallel::RcppParallelLibs()")
```
Comment (TychonautVII) [inline src/Makevars]: Or maybe these new makevars fixed your bug?

File: vignettes/predict-example.Rmd [hunk 3]
```diff
@@ -2,7 +2,7 @@
 title: "Prediction using BCF"
```
Comment (TychonautVII) [inline vignettes/predict-example.Rmd]: Why did you rename these files? I believe they were named this way to have them show up in a specific order

File: src/Makevars (hunk 2 above)
Comment (ignacio82) [reply src/Makevars]: I believe that the change to `Makevars` is what allowed me to install the package on linux and other changes were made to address warnings from `devtools:check`. That said, i did this before my daughter was born and now that i'm sleep deprive my memory is not very reliable...

File: vignettes/predict-example.Rmd [hunk 4]
```diff
@@ -187,13 +187,13 @@ For each unit $j$ in our dataset, for each value of $x_3$ in the bottom/top quar
 #   xPredQ1[,2] <- x20[j,2]
 #   xPredQ4[,2] <- x20[j,2]
 #   # predict the treatment effect for each simulated unit
-#   tauPredQ1 = predict(bcfObj              = bcf_out,
+#   tauPredQ1 = predict(object              = bcf_out,
```
Comment (TychonautVII) [inline vignettes/predict-example.Rmd]: This is another one I'm wonder why it got renamed.

File: vignettes/predict-example.Rmd (hunk 3 above)
Comment (ignacio82) [reply vignettes/predict-example.Rmd]: vignettes can not start with a number (it won't pass the cran check and you won't be able to publish the package)

File: vignettes/predict-example.Rmd (hunk 4 above)
Comment (ignacio82) [reply vignettes/predict-example.Rmd]: I don't remember the right words but i think this is what i want to say: "You are *overloading the method predict, therefore you need to have the same arguments with the same order as the original method"

File: src/Makevars (hunk 2 above)
Comment (TychonautVII) [reply src/Makevars]: Congratulations on your new family member Ignacio! Man, this was such a crazy year for you! I remember that was one of the many variables up in the air, but I didn't put together that your daughter was coming so soon!

File: DESCRIPTION (hunk 1 above)
Comment (ignacio82) [reply DESCRIPTION]: commit message with the relevant link:
https://github.com/ignacio82/bcf/commit/ddf96255963c8986b487fba57103f64d4ecead45

File: vignettes/predict-example.Rmd (hunk 3 above)
Comment (ignacio82) [reply vignettes/predict-example.Rmd]: https://github.com/r-lib/pkgdown/issues/545 

See commit message: https://github.com/ignacio82/bcf/commit/00d51a7c450418cb485a9c8e2e3957070e00530e
